module_bom.go

```diff
@@ -111,3 +145,41 @@ func (m ModuleBOM) Generate(workingDir string) ([]packit.BOMEntry, error) {
 
 	return entries, nil
 }
+
+func getLockFile(workingDir string) (packageLock, error) {
+	file, err := os.Open(filepath.Join(workingDir, "package-lock.json"))
+	if err != nil {
+		return packageLock{}, fmt.Errorf("failed to open package-lock.json: %w", err)
+	}
+	defer file.Close()
+
+	var lockFile packageLock
+	err = json.NewDecoder(file).Decode(&lockFile)
+	if err != nil {
+		return packageLock{}, fmt.Errorf("failed to decode package-lock: %w", err)
+	}
+	return lockFile, nil
+}
+
+// retrieveIntegrityFromLockfile is a function that will read the
+// package-lock.json if there is one, and retrieve the integrity (hash) for a
+// specific dependency. It returns the hash algorithm and hash itself.
+func retrieveIntegrityFromLockfile(packageLock packageLock, pkg string) (string, string, error) {
+	for name, dependency := range packageLock.Dependencies {
+		if name == pkg {
+			dependencyMap := dependency.(map[string]interface{})
+			integrity := dependencyMap["integrity"].(string)
+
+			if strings.Contains(integrity, "-") {
+				algAndHash := strings.Split(integrity, "-")
+				content, err := base64.StdEncoding.DecodeString(algAndHash[1])
+				if err != nil {
+					return "", "", err
+				}
+				return algAndHash[0], hex.EncodeToString(content), nil
+			}
+			return "", "", nil
+		}
+	}
+	return "", "", nil
+}
```

Comment:
These two functions strike me as belonging to the `packageLock` type. Would it make sense for them to be something like the following:
```go
NewPackageLock(workingDir string) (PackageLock, error)
(p PackageLock) FindChecksumFor(pkg string) (packit.BOMChecksum, error)
```